src/todo-gantt/app/Http/Controllers/ProjectController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ProjectRequest;
+use App\Models\Project;
+use App\Models\Team;
+use App\Models\User;
+use Illuminate\Http\Request;
+
+class ProjectController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(ProjectRequest $request)
+    {
+        $user = auth()->user();
+        $current_team = $user->selectedTeam()->first();
```

Comment:
このアクセスはあまり推奨されない書き方ですー。
下記で充分だと思います。
```suggestion
        $current_team = $user->selectedTeam;
```

リレーションシップが設定されている場合、プロパティとしてメソッドを扱えるので、メソッドの呼出と、その結果から first を取得するみたいな書き方をしなくて大丈夫です。
また、これは 1:1 のものだけじゃなく、1:N のものでも同じです (ただし 1:N や N:N のようなケースでは N+1 問題に注意してください)。

Reply:
なるほど！
「プロパティアクセス」と「メソッド呼び出し」の違いが分かっていませんでした；

- 追加のクエリ操作が必要な時は「メソッド呼び出し」
- 不要な場合は「プロパティアクセス」

こんな感じでしょうか？

Reply:
そうですね。なのでメソッド実行をした場合はクエリが実際には未実行のままとなります (クエリビルダを返すので)。
逆に、プロパティでアクセスした場合、その時点でクエリが発行される筈です。

なので `$user->projects->where(...)` みたいな処理をする場合、projects の段階で一度全取得するのでクエリ効率が悪い (か、全取得のデータが大きすぎる) 場合があります。
このケースの場合に where の処理はコレクションとしての処理になるので、プログラムでフィルタする事になります。
`$user->projects()->where(...)` の場合だと、クエリビルダを返しているだけなので、その後に続く where は SQL を構築するためのクエリビルダの処理になるため、取得するデータの制限が正しく動作します。

こういう違いがありますが、プロパティでの取得の方が基本的に読み易く、単純に見えるので基本はプロパティで取得となります。

Reply:
理解しました！